// alea-cards/src/AleaCard.js
import React from 'react';
import PropTypes from 'prop-types';

const AleaCard = ({ image, title, description, order = 'tid', className }) => {
  const components = {
    t: title && <h2 className="alea-card-title">{title}</h2>,
    i: <img src={image} alt={title || 'Card image'} className="alea-card-image" />,
    d: description && <p className="alea-card-description">{description}</p>,
  };

  const orderedComponents = order.split('').map((key) => components[key]);

  return (
    <div className={`alea-card ${className}`} style={styles.card}>
      {orderedComponents}
    </div>
  );
};

AleaCard.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
  order: PropTypes.oneOf(['tid', 'tdi', 'idt', 'itd', 'dti', 'dit']),
  className: PropTypes.string,
};

const styles = {
  card: {
    border: '1px solid #ddd',
    borderRadius: '8px',
    padding: '16px',
    width: '300px',
    textAlign: 'center',
    boxShadow: '0 4px 8px rgba(0,0,0,0.1)',
    margin: '16px',
  },
};

export default AleaCard;
